modules/react-components/src/providers/session-management/session-management-provider.tsx

```diff
@@ -217,19 +260,20 @@ export const SessionManagementProvider: FunctionComponent<PropsWithChildren<
             <SessionTimeoutModal
                 open={ showSessionTimeoutModal }
                 onClose={ handleSessionTimeoutAbort }
-                onPrimaryActionClick={ handleSessionTimeoutAbort }
+                onPrimaryActionClick={ handlePrimaryActionClick }
                 onSecondaryActionClick={ handleSessionLogout }
                 heading={
                     <Trans
-                        i18nKey={ modalOptions?.headingI18nKey }
+                        i18nKey={ !sessionTimedOut?
+                            modalOptions?.headingI18nKey : modalOptions?.sessionTimedOutHeadingI18nKey }
                         tOptions={
                             { time: timerDisplay }
                         }
                     >
                     </Trans>
                 }
-                description={ modalOptions?.description }
-                primaryButtonText={ modalOptions?.primaryButtonText }
+                description={ sessionTimedOut ? modalOptions?.sessionTimedOutDescription : modalOptions?.description }
+                primaryButtonText={ sessionTimedOut ? modalOptions?.loginAgainButtonText : modalOptions?.primaryButtonText }
```

Comment:
Possible max line length issue here.

Reply:
Addressed in https://github.com/wso2/identity-apps/pull/1732/commits/6ce4484d6f7bfe758f7f22f590a45a499e21d508